// Copyright (c) 2021 Systemathics
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

// Defines the intraday sampling intervals.
syntax = "proto3";


package systemathics.apis.type.shared.v1;

// Contains the intraday sampling intervals.
enum Sampling
{
    // The unspecifed interval
    SAMPLING_UNSPECIFIED = 0;
    // The one minute interval
    SAMPLING_ONE_MINUTE = 1;
    // The five minutes interval
    SAMPLING_FIVE_MINUTE = 2;
    // The ten minutes interval
    SAMPLING_TEN_MINUTE = 3;
    // The fifteen minutes interval
    SAMPLING_FIFTEEN_MINUTE = 4;
    // The thirty minutes interval
    SAMPLING_THIRTY_MINUTE = 5;
    // The one hour interval
    SAMPLING_ONE_HOUR = 6;
}
